src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_Call.cs

```diff
@@ -512,6 +490,107 @@ private static bool IsSafeForReordering(BoundExpression expression, RefKind kind
             return actualArguments.AsImmutableOrNull();
         }
 
+        internal static ImmutableArray<BoundExpression> MakeArgumentsInEvaluationOrder(
+            Binder binder,
+            SyntaxNode syntax,
+            ImmutableArray<BoundExpression> arguments,
+            Symbol methodOrIndexer,
+            MethodSymbol optionalParametersMethod,
+            bool expanded,
+            ImmutableArray<int> argsToParamsOpt,
+            bool invokedAsExtensionMethod,
+            /* out */ ArrayBuilder<ArgumentKind> argumentKindBuilder,
+            /* out */ ArrayBuilder<ParameterSymbol> parameterSymbolBuilder,
+            ThreeState enableCallerInfo = ThreeState.Unknown)
+        {
+            Debug.Assert(binder != null && argumentKindBuilder.IsEmpty() && parameterSymbolBuilder.IsEmpty());
+
+            // Either the methodOrIndexer is a property, in which case the method used
+            // for optional parameters is an accessor of that property (or an overridden
+            // property), or the methodOrIndexer is used for optional parameters directly.
+            Debug.Assert(((methodOrIndexer.Kind == SymbolKind.Property) && optionalParametersMethod.IsAccessor()) ||
+                ReferenceEquals(methodOrIndexer, optionalParametersMethod));
+
+            // We need to do a fancy rewrite under the following circumstances:
+            // (1) a params array is being used; we need to generate the array.
+            // (2) there were named arguments that reordered the arguments; we might
+            //     have to generate temporaries to ensure that the arguments are 
+            //     evaluated in source code order, not the actual call order.
+            // (3) there were optional parameters that had no corresponding arguments.
+            //
+            // If none of those are the case then we can just take an early out.
+
+            ImmutableArray<ParameterSymbol> parameters = methodOrIndexer.GetParameters();
+
+            if (IsRewritingRequired(ref arguments, methodOrIndexer, expanded, argsToParamsOpt, invokedAsExtensionMethod, null, out var isComReceiver, out var temporariesBuilder))
```

Comment:
`temporariesBuilder` is not used or freed. If it is always `null`, consider asserting that in the `IsRewritingRequired` code path.